mne/tests/test_source_estimate.py

```diff
@@ -425,21 +425,20 @@ def test_io_stc(tmpdir):
     # test warning for complex data
     stc2.data = stc2.data.astype(np.complex128)
     with pytest.raises(ValueError, match='Cannot save complex-valued STC'):
-        stc2.save(tmpdir.join('complex.stc'))
+        stc2.save(tmp_path / 'complex.stc')
 
 
 @requires_h5py
 @pytest.mark.parametrize('is_complex', (True, False))
 @pytest.mark.parametrize('vector', (True, False))
-def test_io_stc_h5(tmpdir, is_complex, vector):
+def test_io_stc_h5(tmp_path, is_complex, vector):
     """Test IO for STC files using HDF5."""
     if vector:
         stc = _fake_vec_stc(is_complex=is_complex)
     else:
         stc = _fake_stc(is_complex=is_complex)
-    pytest.raises(ValueError, stc.save, tmpdir.join('tmp'),
-                  ftype='foo')
-    out_name = tmpdir.join('tmp')
+    pytest.raises(ValueError, stc.save, tmp_path / 'tmp', ftype='foo')
```

Comment:
```suggestion
    # different errors for (non)vector STCs
    match = (r"(Invalid value for the 'ftype' par)|"
             r"(VectorSourceEstimate objects can only be written ad HDF5)*")
    with pytest.raises(ValueError, match=match):
        stc.save(tmp_path / 'tmp', ftype='foo')
```

Reply:
Should we go with this more specific error message catching or wait for the fix in #9944?

Reply:
I don't think my fix in #9944 will impact this comment -- it looks like this comment just improves/modernizes the `pytest.raises` usage. So yes feel free to commit this suggestion separately